atd-vze/src/views/Crashes/RelatedRecordsTable.js

```diff
@@ -0,0 +1,198 @@
+import React, { useState } from "react";
+import { Table, Badge, Button, Input } from "reactstrap";
+
+import { useAuth0, isReadOnly } from "../../auth/authContext";
+
+const RelatedRecordsTable = ({
+  fieldConfig,
+  data,
+  sortField,
+  tableName,
+  keyField,
+  lookupOptions,
+  mutation,
+  refetch,
+  ...props
+}) => {
+  const [editField, setEditField] = useState("");
+  const [editRow, setEditRow] = useState("");
+  const [formData, setFormData] = useState("");
+
+  // Disable edit features if only role is "readonly"
+  const { getRoles } = useAuth0();
+  const roles = getRoles();
+
+  const handleEditClick = (field, row) => {
+    setEditField(field);
+    setEditRow(row);
+  };
+
+  const handleCancelClick = e => {
+    e.preventDefault();
+
+    // RESET state on cancel
+    setEditField("");
+    setEditRow("");
+    setFormData({});
+  };
+
+  const handleInputChange = e => {
+    // TODO: Make this less specific to person table
```

Comment:
The last three part of the code that need to be refactored to make this component reusable across related tables are these `TODO:`s

- [ ] TODO: Make `handleInputChange` & `handleSubmit` less unspecific to person table
- [ ] TODO: Fix edit bug when persons have same unit id